cppcache/integration-test-2/CommitConflictExceptionTest.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <thread>
+#include <future>
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <geode/CacheTransactionManager.hpp>
+#include <geode/RegionShortcut.hpp>
+#include <geode/RegionFactory.hpp>
+
+#include "framework/Cluster.h"
+
+namespace {
+
+using namespace apache::geode::client;
+using namespace std::chrono;
+
+std::shared_ptr<Region> setupRegion(Cache& cache) {
+  auto region = cache.createRegionFactory(RegionShortcut::PROXY)
+                     .setPoolName("default")
+                     .create("region");
+  return region;
+}
+
+TEST(CommitConflictExceptionTest, putPartitionTx) {
+  Cluster cluster{LocatorCount{1}, ServerCount{1}};
+  cluster.getGfsh()
+      .create()
+      .region()
+      .withName("region")
+      .withType("PARTITION")
+      .execute();
+
+  std::promise<void> begunTx1;
+  std::promise<void> putTx1;
+  std::promise<void> putTx2;
+  std::promise<void> committedTx2;
+
+  auto task1 = std::async(std::launch::async, [&cluster, &begunTx1, &putTx1, &putTx2, &committedTx2] {
+    SCOPED_TRACE("task1");
+    auto cache = cluster.createCache();
+    auto region = setupRegion(cache);
+
+    cache.getCacheTransactionManager()->begin();
+    begunTx1.set_value();
+    region->put("one", "one");
+    ASSERT_EQ(std::future_status::ready,
+              putTx2.get_future().wait_for(debug_safe(minutes(1))));
+    putTx1.set_value();
+    ASSERT_EQ(std::future_status::ready,
+              committedTx2.get_future().wait_for(debug_safe(minutes(1))));
+    try {
```

Comment:
Would `EXPECT_THROW` not work in this case?

Reply:
The problem with EXPECT_THROW or ASSERT_THROW is that i cannot assert that is a CommitConflictException only an apache::geode::client::Exception (at least i couldn't make it work with apache::geode::client::CommitConflictException) and to be sure that is a CommitConflictException i have to examine the content.